components/feature/addons/src/main/java/mozilla/components/feature/addons/amo/AddOnsCollectionsProvider.kt

```diff
@@ -0,0 +1,161 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@file:Suppress("TooManyFunctions")
+
+package mozilla.components.feature.addons.amo
+
+import mozilla.components.concept.fetch.Client
+import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.isSuccess
+import mozilla.components.feature.addons.AddOnsDiscoverabilityProvider
+import mozilla.components.feature.addons.AddOn
+import org.json.JSONArray
+import org.json.JSONObject
+import java.io.IOException
+
+internal const val API_VERSION = "api/v4"
+internal const val DEFAULT_SERVER_URL = "https://addons.mozilla.org"
+internal const val DEFAULT_COLLECTION_NAME = "7e8d6dc651b54ab385fb8791bf9dac"
+
+/**
+ * Provide access to the collections AMO API.
+ * https://addons-server.readthedocs.io/en/latest/topics/api/collections.html
+ *
+ * @property serverURL The URL of the server which you want to interact e.g production, staging
```

Comment:
nit: `serverUrl` The `Url`of the endpoint to interact with e.g. a prouction, testing or staging server.